src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -154,13 +154,15 @@ private async Task DoReceive()
 
         private async Task ProcessReceives()
         {
+            // Resolve `input` PipeWriter via the IDuplexPipe interface prior to loop start
+            var input = Input;
```

Comment:
Nit: I would mention in the comment that this is for perf not correctness.

Reply:
Done